Automattic/jetpack
# Intro tour for classic admin interface

Fixes https://github.com/Automattic/dotcom-forge/issues/7404

## Proposed changes:

Adds a dismissible intro tour which displays some educational tooltips for folks who turn the classic admin interface on:

Step 1 | Step 2 | Step 3
--- | --- | ---
<img width="1280" alt="Screenshot 2024-05-27 at 14 45 19" src="https://github.com/Automattic/jetpack/assets/1233880/7e8f823c-2a1c-4ee5-908f-5e3efdfc72f9"> | <img width="1280" alt="Screenshot 2024-05-27 at 14 45 25" src="https://github.com/Automattic/jetpack/assets/1233880/ca025847-1972-4d29-83ae-fd3140be22f6"> | <img width="1277" alt="Screenshot 2024-05-27 at 14 45 32" src="https://github.com/Automattic/jetpack/assets/1233880/c519faf0-03ef-4476-a928-378fa3f38103">

The tour shows up in the WP Admin dashboard when the `?admin-interface-changed=true` query param is present, which (as of https://github.com/Automattic/wp-calypso/pull/91102) is automatically added to the URL after activating the classic admin interface.

We also display a success notification to indicate that the admin interface has changed:

<img width="500" alt="Screenshot 2024-05-27 at 14 47 30" src="https://github.com/Automattic/jetpack/assets/1233880/5c23a634-94f9-4383-8ec9-b3810ed9d216">

**Some considerations** 
- The intro tour is hidden on mobile viewports
- The tooltips always have the same dark color regardless of the color scheme selected on the site
- The tooltips are not repositioned if the sidebar or the site management panel card are collapsed. The effort needed to address this is probably not worth the number of users who would face this issue
- The copy of the tour is being discussed here: pfsHM7-17R-p2


### Other information:


## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
- Apply these changes to your WP.com testing site
- Go to wordpress.com/settings/general and select your site
- Change the admin interface to Classic
- You should land in `/wp-admin/?admin-interface-changed=true`
- Make sure the intro tour shows up
- Make sure you can proceed to next steps
- Make sure they are placed correctly
- Make sure you the tour is dismissed after clicking on the last step's CTA (or after clicking on the X button that appears in every step)
- Make sure the tour no longer shows up after being dismissed (the success notification should still be displayed)

If you need the tour to be visible after dismissing it, delete the `wpcom_classic_tour_completed` option.
